src/PackageVersions/Installer.php

```diff
@@ -177,6 +177,6 @@ private static function getVersions(Locker $locker, RootPackageInterface $rootPa
             );
         }
 
-        yield $rootPackage->getName() => $rootPackage->getVersion() . '@' . $rootPackage->getSourceReference();
+        yield $rootPackage->getName() => ($rootPackage->getPrettyVersion() ?? $rootPackage->getVersion()) . '@' . $rootPackage->getSourceReference();
```

Comment:
Which composer API introduced `getPrettyVersion`? Can you please make sure the constraints are correct for that to work?

Reply:
Sorry, I didn't check that because I thought that the `--prefer-lowest` of the CI covered that. I've tracked down the addition of that method to this commit from 2011: https://github.com/composer/composer/commit/8e6f8ae57efce1ec0716a5e897a08e500d74f2e5#diff-d65a9c0319702b1659e4d5adce449e79

But I cannot find a source for the versioning of the API... where is it stored?

Reply:
Should be https://github.com/Ocramius/PackageVersions/blob/master/composer.json#L14

Reply:
Sorry, I didn't explain me well... I don't know where I can find some docs or reference for that API so I can track down the corresponding version to that Composer commit!

Reply:
The plugin API version seems to have been declared here in 2013:

https://github.com/composer/composer/commit/92b1ee2f7ad1c4c823acd96b37dd1986f00bdf02#diff-9761e5e6cef580d24c5a05d870dcc479R30

So we should be ok! :+1:

Reply:
:+1:

Reply:
@Jean85 what's the case when you expect `$rootPackage->getPrettyVersion()` to return `NULL` so that the code falls back to using `$rootPackage->getVersion()`? Given that the pretty version is the one obtained from the VCS, it should be always available. Otherwise, there's nothing to be further normalized.

If I'm wrong, and it's possible then this case probably should be covered by an additional test.

Reply:
I've added that because it's already covered by some test somewhere. If you try to remove that, you will see PHPUnit fail.

Reply:
Hmm... that test may as well be wrong then. Can you dig it up so we review it?

Reply:
I've commited the differences, see the latest commit of this PR: ca0fd61552d95a2cd03a07b5769af9188eeb0a41

I just had to update the mocks! I'll commit a small refactoring of the tests now...